dask/dataframe/utils.py

```diff
@@ -596,9 +596,8 @@ def assert_dask_graph(dask, label):
             k = k[0]
         if k.startswith(label):
             return True
-    else:
```

Comment:
no `break` in the `for` loop, so `else` is unnecessary/ typo "contan" -> "contain" in the `AssertionError`

Reply:
for-else is also valid if the iterator is falsey

```python
In [1]: for i in []:
   ...:     pass
   ...: else:
   ...:     print('hello')
   ...:     
hello
```

Reply:
Maybe I'm missing something, but isn't the `else` still redundant in that case?  My understanding is that the `else` block of a `for-else` will always execute upon completion of the `for` loop, unless the `for` loop is explicitly broken early via `break` (a falsey iterator would trivially complete without `break`).  So, the only case `else` is strictly necessary is if there's a possibility of a `break` interrupting the loop; if not, it will just be equivalent to running the `else` block code after the loop without an `else` clause (i.e. remove the `else:` and unindent).  Is that incorrect?  Happy to leave it as a `for-else` if that's something you'd prefer stylistically.

Reply:
Indeed.  I just checked things and my understanding of for-else was incorrect.

Reply:
No worries.  Thanks for rerunning Travis (I'm guessing that was you?).  I didn't think my changes would have caused that memory allocation error, but wanted to triple check things before trying to kick off CI again since I'm not too familiar with the codebase yet.